crates/api/src/wire.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package kitsune2.wire;
+
+// A Kitsune2 wire protocol message.
+//
+// This is the top-level encoding
+// that will be transferred between Kitsune2 peers. Most communications
+// between peers to make Kitsune2 actually function will be encoded
+// separately inside the payload of TY_MODULE type messages.
+message K2Proto {
+  // Enumeration of the types of messages that can be sent between peers.
+  //
+  // We are using this enum field to distinguish between top-level messages,
+  // rather than protobuf's oneof because of the downsides of upgrading oneofs.
+  enum Ty {
+    // The "UNSPECIFIED" type for future message types.
+    //
+    // In general, peers should ignore unspecified messages, but
+    // should still count them toward any ratelimiting metrics.
+    TY_UNSPECIFIED = 0;
+
+    // This message is preflight data.
+    //
+    // The implementor is responsible for encoding any module-specific
+    // requirements within the data payload of this message type.
+    // For example, if peers are required to include the "dht_v1" module
+    // to communicate with each other, they should reject preflight to
+    // peers that do not include that module.
+    TY_PREFLIGHT = 1;
+
+    // This is a notification or fire-and-forget message from a peer.
+    //
+    // This type requires that a "space" be specified.
+    TY_NOTIFY = 2;
```

Comment:
I named this NOTIFY because, especially in the protobuf domain, a "message" is something already. But in my other space PR i've got "incoming_message" and "send_message" which would use this type. So, need some naming help here.

Reply:
Based on the fact that they are "fire-and-forget" then I think that notification makes sense to me so how about just:
```suggestion
    TY_NOTIFICATION = 2;
```

Reply:
I'm happy with NOTIFY, as long as the naming in method names matches